src/GitVersion.Core/VersionCalculation/TrunkBased/NonTrunk/CommitOnNonTrunkBranchedBase.cs

```diff
@@ -0,0 +1,29 @@
+using GitVersion.Configuration;
+using GitVersion.Extensions;
+
+namespace GitVersion.VersionCalculation.TrunkBased.NonTrunk;
+
+internal abstract class CommitOnNonTrunkBranchedBase : ITrunkBasedIncrementer
+{
+    public virtual bool MatchPrecondition(TrunkBasedIteration iteration, TrunkBasedCommit commit, TrunkBasedContext context)
+        => !commit.Configuration.IsMainline && commit.BranchName != iteration.BranchName && commit.Successor is null;
+
+    public virtual IEnumerable<BaseVersionV2> GetIncrements(TrunkBasedIteration iteration, TrunkBasedCommit commit, TrunkBasedContext context)
+    {
+        context.BaseVersionSource = commit.Value;
+
+        var incrementForcedByBranch = iteration.Configuration.Increment == IncrementStrategy.Inherit
+            ? commit.GetIncrementForcedByBranch() : iteration.Configuration.Increment.ToVersionField();
+        context.Increment = context.Increment.Consolidate(incrementForcedByBranch);
+
+        context.Label = iteration.Configuration.GetBranchSpecificLabel(iteration.BranchName, null) ?? context.Label;
+        context.ForceIncrement = true;
+
+        yield return BaseVersionV2.ShouldIncrementFalse(
+            source: GetType().Name,
```

Comment:
Please replace `GetType().Name` with `nameof(}` to have it set at compile time instead of runtime.

Please update everywhere the source is set for `BaseVersionV2.ShouldIncrementFalse`

Reply:
Actually because of the nature of the base class I would say to use the underlying type instead of the base class type. If I use nameof the name of the base class will be used not the name of the derived class.

Are you sure to change it to nameof?

Reply:
I think for those that do override the `GetIncrements` method like in the `LastCommit*` methods we should use the `nameof`, and find a way to avoid using the refection. At some point we will need to make it AOT compatible so better to avoid introducing reflection.

Reply:
Let's go with the current version. We can later improve this